### PR TITLE
BC-2340 Task import and sharing

### PR DESCRIPTION
**Pull request by MateuszSapalaPassCon**
# Description
This change will add lesson import and sharing.

QA note: No Issues Found. See file on ticket.

## Links to Tickets or other pull requests
- https://ticketsystem.dbildungscloud.de/browse/BC-2340
- https://github.com/hpi-schul-cloud/dof_app_deploy/pull/466
- https://github.com/hpi-schul-cloud/schulcloud-server/pull/3897

## Approval for review

- [x] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### CODE REVIEW

**Review by VicDavydiuk — APPROVED** [review]
Looks good

File: src/locales/de.json [hunk 1]
```diff
@@ -511,6 +511,10 @@
 	"components.molecules.share.lesson.result.linkLabel": "Link Themakopie",
 	"components.molecules.share.lesson.mail.subject": "Thema zum Importieren",
 	"components.molecules.share.lesson.mail.body": "Link zum Thema:",
+	"components.molecules.share.task.options.infoText": "Mit dem folgenden Link kann die Aufgabe als Kopie von anderen Lehrkräften importiert werden. Personenbezogene Daten werden dabei nicht importiert.",
+	"components.molecules.share.task.result.linkLabel": "Link Aufgabekopie",
+	"components.molecules.share.task.mail.subject": "Aufgabe zum Importieren",
+	"components.molecules.share.task.mail.body": "Link zum Aufgabe:",
```
Comment (hoeppner-dataport) [inline src/locales/de.json:517]: Only a minor language thing: "Link zur Aufgabe:", zum => zur

File: src/store/share-task.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
+import { $axios } from "@/utils/api";
```
Comment (hoeppner-dataport) [inline src/store/share-task.ts:1]: Did you consider to have a general ShareModule for courses, tasks and lessons?
They all look very similar and use "shareTokenControllerCreateShareToken"...

File: src/store/share.ts [hunk 3]
```diff
@@ -36,11 +42,12 @@ export default class ShareCourseModule extends VuexModule {
...
 	): Promise<ShareTokenResponse | undefined> {
 		const shareTokenPayload: ShareTokenBodyParams = {
-			parentType: ShareTokenBodyParamsParentTypeEnum.Courses,
-			parentId: this.courseId,
+			parentType: this.parentType,
+			parentId: this.parentId,
 			expiresInDays: payload.hasExpiryDate ? 21 : null,
 			schoolExclusive: payload.isSchoolInternal,
 		};
+		console.log(shareTokenPayload);
```
Comment (hoeppner-dataport) [inline src/store/share.ts:50]: You left some console.logs in this file. There is at least one other one below.

File: src/store/share-task.ts (hunk 2 above)
Comment (MateuszSapalaPassCon) [reply src/store/share-task.ts]: I hadn't thought of that, but it was a good suggestion and I changed to one module so as not to duplicate code

**Review by hoeppner-dataport — APPROVED** [review]

**Review by hoeppner-dataport — DISMISSED** [review]

**Review by VikDavydiuk — APPROVED** [review]

**Pull request merged**